src/opam.ml

```diff
@@ -127,14 +129,30 @@ end = struct
 
   let of_switch opam switch =
     let open Promise.Syntax in
-    let path = Switch.path opam switch in
-    let switch_state_filepath = Opam_path.switch_state path |> Path.to_string in
-    let+ file_content = Fs.readFile switch_state_filepath in
-    let { OpamParserTypes.file_contents; _ } =
-      OpamParser.FullPos.string file_content switch_state_filepath
-      |> OpamParser.FullPos.to_opamfile
+    let switch_state_filepath =
+      Switch.path opam switch |> Opam_path.switch_state |> Path.to_string
     in
-    file_contents
+    let+ file_res =
+      Promise.Result.from_catch (Fs.readFile switch_state_filepath)
+    in
+    match file_res with
+    | Error e ->
+      log "Error reading switch-state file at %s. Error: %s"
+        switch_state_filepath
+        (Promise.error_to_js e |> Ojs.string_of_js);
```

Comment:
Does `Fs.readFile` throw strings as exceptions on the JS side?
I think it is more appropriate to read the message with `JsError.message`.

Reply:
That's a good point. But then 

1. Did you mean `Js_of_ocaml.Js_error.message`? 
2. How does one go from `Promise.error` to `Js_error.t`?

Reply:
In case you're interested: based on this typescript [intf](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7f7fef74f8c7116816392e6ba1d873203babb7cc/types/mz/fs.d.ts#L1132-L1136), the exception thrown should be [NodeJs.ErrornoException](http://definitelytyped.org/docs/pg--pg/interfaces/nodejs.errnoexception.html) 

I wonder if having a safe wrapper over such functions would prove useful, efficient, and maintainable, eg `readFile: path:string -> (string, NodeJs.ErrornoException.t) result Promise.t`

Reply:
> 1. Did you mean `Js_of_ocaml.Js_error.message`?
> 
> 2. How does one go from `Promise.error` to `Js_error.t`?

No, I mean `Node.JsError.message` from our own bindings. `JsError.t` is `Promise.error`:
https://github.com/ocamllabs/vscode-ocaml-platform/blob/0fc1e0d59c7d406c410d283146950c9a543bf4bb/src-bindings/node/node.mli#L41-L45

We already use this function to get the message in other places:
https://github.com/ocamllabs/vscode-ocaml-platform/blob/0fc1e0d59c7d406c410d283146950c9a543bf4bb/src/extension_instance.ml#L234-L236

> I wonder if having a safe wrapper over such functions would prove useful, efficient, and maintainable

It could be useful in the future, but I don't think it's necessary here since we only need the message for logging purposes.

Reply:
Thanks!